File: ex_448.cpp
#include <algorithm>
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    vector<int> findDisappearedNumbers(vector<int>& nums)
    {
        if (nums.empty())
            return {};
        int len = nums.size();
        vector<int> vec(len + 1);
        for (auto i : nums)
        {
            vec[i]++;
        }
        vector<int> res;
        for (int i = 1; i < len + 1; ++i)
        {
            if (vec[i] == 0)
                res.emplace_back(i);
        }
        return res;
    }
};

int main(void)
{
    Solution sy;
    vector<int> nums = {1, 3, 4, 5, 3, 6, 8};
    auto result = sy.findDisappearedNumbers(nums);
    ostream_iterator<int> output(cout, " ");
    copy(result.cbegin(), result.cend(), output);
    cout << endl;

    return 0;
}

File: ex_448.py
#!/usr/bin/env python3

class Solution:
    # def findDisappearedNumbers(self, nums):
    # exam = set()
    # for i in nums:
    # exam.add(i)
    # res = []
    # for i in range(1, len(nums)+1):
    # if i not in exam:
    # res.append(i)
    # return res
    def findDisappearedNumbers(self, nums):
        return list(set([i for i in range(1, len(nums)+1)]) ^ set(nums))


test = [1, 3, 3, 4, 6, 8]
sy = Solution()

print(sy.findDisappearedNumbers(test))
